### DISCUSSION TITLE
Build a list of "WHERE ... OR ..." from a List of Expressions

### DISCUSSION THREAD

**Discussion by gsouf** [simolus3/drift#2642] [Q&A] [answered]
Hi,

I'm trying to build a where statement for my query, that is a bunch of OR conditions. Typically I would use the `where` method and the operator `|` in this way:

```dart
query.where(table.column1.eq(1) | table.column2.eq(1));
```

Now I'm trying to do that dynamically from a List where I can have zero or many Expression.
If I had to connect these conditions with "AND" operator it would have been as easy as iterating over the list and calling `where` many time.

However finding a way to get it done with "OR" is puzzling me. This is how I did it, but I'm wondering if drift comes with a better way of doing it:

```dart
List<Expression<bool>> myExpressions = [
   table.column1.eq(1),
   table.column2.eq(1),
];


if (myExpressions.isNotEmpty) {
  // get first expression
  Expression<bool> orExpressions = myExpressions.removeAt(0);

  // connect all other expressions with | operator
  for(Expression<bool> newExpression in myExpressions) {
    orExpressions = orExpressions | newExpression;
  }
  
  // add to the query
  query.where(orExpressions);
}

```

Any idea?
Thanks

**Comment by simolus3** [comment, marked as answer]
It's not built into drift, but you can use `Iterable.reduce` to express this concisely:

```dart
if (myExpressions.isNotEmpty) {
  query.where(myExpressions.reduce((value, element) => value | element));
}
```

**Reply by gsouf** [reply]
That's a good one thanks. Do you think it would be a good feature to add, or is it too specific? 

I'm thinking for example of something like a factory or something like that, i.e:   `Expression.or(myExpressions)`.

**Reply by simolus3** [reply]
I think that would be useful. I'm just not sure how we deal with an empty iterable though. Mathematically, it makes sense for an empty disjunction to be false but in your case it should match all rows instead. But maybe we should just make that the default and allow a parameter for the default value.
